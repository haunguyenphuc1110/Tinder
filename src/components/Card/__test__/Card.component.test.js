import React from 'react';
import renderer from 'react-test-renderer';

import Items from '../Card.component';

it('renders correctly when there are no items', () => {
  const props = {
    panHandlers: jest.fn(), 
    rotateAndTranslate: jest.fn(), 
    likeOpacity: jest.fn(), 
    dislikeOpacity: jest.fn()
  }
  const tree = renderer.create(<Items {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});