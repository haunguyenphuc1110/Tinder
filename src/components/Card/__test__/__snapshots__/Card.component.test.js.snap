// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<View
  _isMockFunction={true}
  getMockImplementation={[Function]}
  getMockName={[Function]}
  mock={
    Object {
      "calls": Array [],
      "instances": Array [],
      "invocationCallOrder": Array [],
      "results": Array [],
    }
  }
  mockClear={[Function]}
  mockImplementation={[Function]}
  mockImplementationOnce={[Function]}
  mockName={[Function]}
  mockRejectedValue={[Function]}
  mockRejectedValueOnce={[Function]}
  mockReset={[Function]}
  mockResolvedValue={[Function]}
  mockResolvedValueOnce={[Function]}
  mockRestore={[Function]}
  mockReturnThis={[Function]}
  mockReturnValue={[Function]}
  mockReturnValueOnce={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f8f9fa",
      "borderRadius": 5,
      "elevation": 10,
      "flex": 1,
      "height": 1134,
      "justifyContent": "center",
      "margin": 10,
      "padding": 10,
      "position": "absolute",
      "width": 730,
    }
  }
>
  <View
    style={
      Object {
        "left": 20,
        "opacity": [MockFunction],
        "position": "absolute",
        "top": 30,
        "transform": Array [
          Object {
            "rotate": "-30deg",
          },
        ],
        "zIndex": 1000,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#186320",
          "borderWidth": 1,
          "color": "#186320",
          "fontSize": 32,
          "fontWeight": "800",
          "padding": 10,
        }
      }
    >
      LIKE
    </Text>
  </View>
  <View
    style={
      Object {
        "opacity": [MockFunction],
        "position": "absolute",
        "right": 20,
        "top": 30,
        "transform": Array [
          Object {
            "rotate": "30deg",
          },
        ],
        "zIndex": 1000,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#b30000",
          "borderWidth": 1,
          "color": "#b30000",
          "fontSize": 32,
          "fontWeight": "800",
          "padding": 10,
        }
      }
    >
      NOPE
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#f8f9fa",
        "borderTopWidth": 1,
        "bottom": 0,
        "height": 759.3333333333334,
        "justifyContent": "center",
        "position": "absolute",
        "width": 730,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "gray",
          "borderRadius": 100,
          "borderWidth": 1,
          "left": 290,
          "padding": 4,
          "position": "absolute",
          "top": -100,
          "zIndex": 9999,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "",
          }
        }
        style={
          Object {
            "borderRadius": 100,
            "height": 150,
            "resizeMode": "cover",
            "width": 150,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginTop": 80,
        }
      }
    >
      My address is
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 24,
          "fontWeight": "400",
        }
      }
    >
      4661 Auburn Ave
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-evenly",
          "marginTop": 40,
          "width": 250,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#ffffff",
                "borderWidth": 1,
                "height": 1,
                "marginBottom": 5,
                "width": 30,
              },
              Object {
                "borderColor": "#ffffff",
              },
            ]
          }
        />
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 28,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#ffffff",
                "borderWidth": 1,
                "height": 1,
                "marginBottom": 5,
                "width": 30,
              },
              Object {
                "borderColor": "#ffffff",
              },
            ]
          }
        />
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 28,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Regular",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "400",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#ffffff",
                "borderWidth": 1,
                "height": 1,
                "marginBottom": 5,
                "width": 30,
              },
              Object {
                "borderColor": "#186320",
              },
            ]
          }
        />
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#186320",
                "fontSize": 28,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#ffffff",
                "borderWidth": 1,
                "height": 1,
                "marginBottom": 5,
                "width": 30,
              },
              Object {
                "borderColor": "#ffffff",
              },
            ]
          }
        />
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 28,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#ffffff",
                "borderWidth": 1,
                "height": 1,
                "marginBottom": 5,
                "width": 30,
              },
              Object {
                "borderColor": "#ffffff",
              },
            ]
          }
        />
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 28,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
